sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -1353,9 +1353,21 @@ void simd_obj_impl<T, N, T1, SFINAE>::copy_to(
       if constexpr (RemN == 1) {
         Addr[NumChunks * ChunkSize] = Tmp[NumChunks * ChunkSize];
       } else if constexpr (RemN == 8 || RemN == 16) {
-        simd<uint32_t, RemN> Offsets(0u, sizeof(T));
-        scatter<UT, RemN>(Addr + (NumChunks * ChunkSize), Offsets,
-                          Tmp.template select<RemN, 1>(NumChunks * ChunkSize));
+        // TODO: Remove once fixed on GPU side
+        if constexpr (sizeof(T) == 1 && RemN == 16) {
+          simd<int32_t, 4> Vals;
+          Vals.template select<4, 1>() =
+              Tmp.template bit_cast_view<int32_t>().template select<4, 1>(
+                  NumChunks * ChunkSize);
+          block_store<int32_t, 4>(reinterpret_cast<int32_t *>(Addr) +
```

Comment:
I realized that using block_store requires the address to be oword (16-byte) aligned, 
while user is not limited with such restriction and can use any pointer in simd::store_to() operation.
The alignment in this block here may be any and thus block_store is risky/incorrect.
Please fix me if there is something that guarantees safety for block_store here.